src/nodes.coffee

```diff
@@ -5348,34 +5353,61 @@ unfoldSoak = (o, parent, name) ->
   ifn
 
 # Constructs a string or regex by escaping certain characters.
-makeDelimitedLiteral = (body, options = {}) ->
-  body = '(?:)' if body is '' and options.delimiter is '/'
+makeDelimitedLiteral = (body, {delimiter: delimiterOption, escapeNewlines, double, includeDelimiters = yes, escapeDelimiter = yes} = {}) ->
+  body = '(?:)' if body is '' and delimiterOption is '/'
+  escapeTemplateLiteralCurlies = delimiterOption is '`'
   regex = ///
       (\\\\)                               # Escaped backslash.
     | (\\0(?=[1-7]))                       # Null character mistaken as octal escape.
-    | \\?(#{options.delimiter})            # (Possibly escaped) delimiter.
-    | \\?(?: (\n)|(\r)|(\u2028)|(\u2029) ) # (Possibly escaped) newlines.
+    #{
+      if escapeDelimiter
+        "|\\\\?(#{delimiterOption})"       # (Possibly escaped) delimiter.
+      else
+        ''
+    }
+    #{
+      if escapeTemplateLiteralCurlies
+        "|\\\\?(\\$\\{)"                   # `${` inside template literals must be escaped.
+      else
+        ''
+    }
+    | \\?(?:
+        #{if escapeNewlines then '(\n)|' else ''}
+          (\r)
+        | (\u2028)
+        | (\u2029)
+      )                                    # (Possibly escaped) newlines.
     | (\\.)                                # Other escapes.
   ///g
-  body = body.replace regex, (match, backslash, nul, delimiter, lf, cr, ls, ps, other) -> switch
-    # Ignore escaped backslashes.
-    when backslash then (if options.double then backslash + backslash else backslash)
-    when nul       then '\\x00'
-    when delimiter then "\\#{delimiter}"
-    when lf        then '\\n'
-    when cr        then '\\r'
-    when ls        then '\\u2028'
-    when ps        then '\\u2029'
-    when other     then (if options.double then "\\#{other}" else other)
-  "#{options.delimiter}#{body}#{options.delimiter}"
+  body = body.replace regex, (match, backslash, nul, ...args) ->
+    delimiter =
+      args.shift() if escapeDelimiter
```

Comment:
I personally like this syntax (as opposed to single-line `delimiter = args.shift() if escapeDelimiter` which means something different) but if it makes you uneasy I could rewrite these as eg
```
delimiter =
  if escapeDelimiter
    args.shift()
  else
    null
```